Assets/Scripts/MRCCameraServer.cs

```diff
@@ -0,0 +1,173 @@
+    private void ListenForIncommingRequests () {         
+        try {                        
+            tcpListener = new TcpListener(IPAddress.Any, 1337);             
+            tcpListener.Start();              
+            Debug.Log("[CAMERA SERVER] Server is listening");              
+            
+            byte[][] receivedBuffers = { new byte[MRCCameraServer.MaxBufferLength], new byte[MRCCameraServer.MaxBufferLength] };
+            int receivedBufferIndex = 0;
+            int receivedBufferDataSize = 0;
+
+            while (true) {                 
+                using (connectedTcpClient = tcpListener.AcceptTcpClient()) {
+
+                    using (NetworkStream stream = connectedTcpClient.GetStream()) {                         
+                        int length; 
+
+                        int maximumDataSize = MRCCameraServer.MaxBufferLength - receivedBufferDataSize;
+
+                        while ((length = stream.Read(receivedBuffers[receivedBufferIndex], receivedBufferDataSize, maximumDataSize)) != 0) {                         
+
+                            receivedBufferDataSize += length;
+
+                            while (receivedBufferDataSize >= CameraPayload.StructSize) {
+                                CameraPayload payload = CameraPayload.FromBytes(receivedBuffers[receivedBufferIndex]);
+
+                                if (payload.protocolIdentifier != CameraPayload.identifier) {
+                                    Debug.LogWarning("Header mismatch");
+                                    stream.Close();
+                                    connectedTcpClient.Close();
+                                    return;
+                                }
+
+                                // Consider adding a lock
+                                cameraPose = payload.ToPose();
+
+                                int newBufferIndex = 1 - receivedBufferIndex;
+                                int newBufferDataSize = receivedBufferDataSize - CameraPayload.StructSize;
+
+                                if (newBufferDataSize > 0) {
+                                    Array.Copy(receivedBuffers[receivedBufferIndex], CameraPayload.StructSize, receivedBuffers[newBufferIndex], 0, newBufferDataSize);
+                                }
+                                receivedBufferIndex = newBufferIndex;
+                                receivedBufferDataSize = newBufferDataSize;
+                                maximumDataSize = MRCCameraServer.MaxBufferLength - receivedBufferDataSize;
+                            }                    
+                        }                                             
+                    }                 
+                }             
+            }         
+        }         
+        catch (SocketException socketException) {             
+            Debug.Log("SocketException " + socketException.ToString());         
+        }     
+    }
+}
```

Comment:
This is the server that receives camera pose updates from the [Reality Mixer](https://github.com/fabio914/RealityMixer) app, you can find the related changes here: https://github.com/fabio914/RealityMixer/pull/44